client-v2/src/app/store/user/user.effects.ts

```diff
@@ -0,0 +1,81 @@
+import { Injectable } from '@angular/core'
+import { Router } from '@angular/router'
+import { HotToastService } from '@ngneat/hot-toast'
+import { Actions, createEffect, ofType } from '@ngrx/effects'
+import { catchError, map, mergeMap, Observable, of, tap } from 'rxjs'
+import { HttpServerErrorResponse } from 'src/app/http/types'
+import { UserService } from 'src/app/services/user.service'
+import { appActions } from '../app.actions'
+import { userActions } from './user.actions'
+import { AuthSuccessResponse, SignupCredentialsDto } from './user.model'
+
+@Injectable()
+export class UserEffects {
+    constructor(
+        private actions$: Actions,
+        private userService: UserService,
+        private toast: HotToastService,
+        private router: Router
+    ) {}
+
+    loginOrSignup = createEffect(() => {
+        return this.actions$.pipe(
+            ofType(userActions.login, userActions.signup),
+            mergeMap(({ type, ...credentials }) => {
+                const isLogin = /login/.test(type)
+                const res$: Observable<AuthSuccessResponse> = isLogin
+                    ? this.userService.login(credentials)
+                    : this.userService.signup(credentials as SignupCredentialsDto)
+
+                return res$.pipe(
+                    this.toast.observe({
+                        success: res => res.successMessage,
+                        error: (err: HttpServerErrorResponse) =>
+                            err.error.message instanceof Array ? err.error.message[0] : err.error.message,
```

Comment:
We should remove the potential key hints (`email: ...`) before showing the toasts